Java/jni/native_fleece.cc

```diff
@@ -339,7 +340,20 @@ Java_com_couchbase_litecore_fleece_FLValue_asDouble(JNIEnv *env, jclass clazz, j
 JNIEXPORT jstring JNICALL
 Java_com_couchbase_litecore_fleece_FLValue_asString(JNIEnv *env, jclass clazz, jlong jvalue) {
     FLString str = FLValue_AsString((FLValue) jvalue);
-    return toJString(env, {str.buf, str.size});
+    const char* newBytes;
+    size_t len = UTF8ToModifiedUTF8((const char*)str.buf, &newBytes, str.size);
+    if(len == 0) {
```

Comment:
What if it's an empty string? Then returning 0 is success, not failure.
(Make sure the unit tests include an empty string!)